packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -416,17 +421,33 @@ class VswhereDetails {
 
     return VswhereDetails(
       meetsRequirements: meetsRequirements,
-      installationPath: details['installationPath'] as String?,
-      displayName: details['displayName'] as String?,
-      fullVersion: details['installationVersion'] as String?,
+      installationPath: _readString(details['installationPath']),
+      displayName: _readString(details['displayName']),
+      fullVersion: _readString(details['installationVersion']),
       isComplete: details['isComplete'] as bool?,
       isLaunchable: details['isLaunchable'] as bool?,
       isRebootRequired: details['isRebootRequired'] as bool?,
       isPrerelease: details['isPrerelease'] as bool?,
-      catalogDisplayVersion: catalog == null ? null : catalog['productDisplayVersion'] as String?,
+      catalogDisplayVersion: catalog == null ? null : _readString(catalog['productDisplayVersion']),
     );
   }
 
+  static String? _readString(dynamic value) {
```

Comment:
nit: I'd be tempted to make this `_validateString(String? value)` and move the casts back up above where the other casts are, just for consistency.

Reply:
Done!